src/server/server.ts

```diff
@@ -48,6 +56,56 @@ function isOfficeHost(host: string) {
 }
 
 
+let generatedDirectories = {};
+/* Cleans up generated template directories */
+setInterval(() => {
+    let ids = Object.keys(generatedDirectories);
+    for (let id of ids) {
+        let directoryInfo = generatedDirectories[id];
+        let now = (new Date()).getTime();
+        // If generated file is more than an hour old and isn't being read, delete it from both server and map to clean up clutter
+        if (directoryInfo && !directoryInfo.isBeingRead && (now - directoryInfo.timestamp) > ONE_HOUR_MS) {
+            delete generatedDirectories[id];
+            fs.unlink(directoryInfo.name, (err) => {
+                if (err) {
+                    ai.trackEvent('File deletion failed', { name: directoryInfo.name });
+                    return;
+                }
+            });
+        }
+    }
+}, ONE_HOUR_MS);
+
+/* Cleans up lurker template directories that may not have been deleted by above function (perhaps due to server restart) */
+setInterval(() => {
+    let filenames: string[] = [];
+    for (let id of Object.keys(generatedDirectories)) {
```

Comment:
Cleaner to write it:
Object.values(generatedDirectories).map((directoryInfo)=> {directoryInfo.relativeFilePath})

Reply:
Object.values(generatedDirectories).map((directoryInfo)=> directoryInfo.relativeFilePath)

Reply:
Discussed offline. Object.values is not supported, so will stick with Object.keys